lib/angular/fieldLocation/fieldLocationController.js

```diff
@@ -12,22 +18,53 @@ function FieldLocationController($scope) {
   //TODO Repeating Fields
   fieldLocationCtrl.field.value = fieldLocationCtrl.field.values[0];
 
+  /**
+  * Attempt to set the current position in the input fields if the current position can be obtained.
+  * Otherwise show a relevant error message to the user.
+  */
   fieldLocationCtrl.setLocation = function(event) {
     event.stopPropagation();
 
-    //TODO Move it to a service to get the location
-    navigator.geolocation.getCurrentPosition(function(pos) {
+    /*
+    * Get the current position from the Geolocation Service.
+    */
+    GeolocationService.getGeolocation(function(err, response) {
+      if (err) {
+        $scope.$apply(function() {
+          // return a relevant error message in the UI
+          fieldLocationCtrl.field.err = err.message ? err.message : err;
+        });
+        return;
+      }
       $scope.$apply(function() {
-        fieldLocationCtrl.field.value.lat = parseFloat(pos.coords.latitude);
-        fieldLocationCtrl.field.value.long = parseFloat(pos.coords.longitude);
-        console.log('position set', fieldLocationCtrl.field.value);
+        // set the longitude/latitude
+        fieldLocationCtrl.field.value.lat = response.coords.latitude;
+        fieldLocationCtrl.field.value.long = response.coords.longitude;
       });
-    }, function() {
-      console.log("Location Not available.");
-      fieldLocationCtrl.field.value.lat = -1;
-      fieldLocationCtrl.field.value.long = -1;
     });
+
+    /**
+    * Apply the longitude/latitude on the scope.
+    * @param latitude - the latitude to set
+    * @param longitude - the longitude to set
+    */
+    fieldLocationCtrl.applyValues = function(latitude, longitude) {
+      $scope.$apply(function() {
+        fieldLocationCtrl.field.value.lat = parseFloat(latitude);
+        fieldLocationCtrl.field.value.long = parseFloat(longitude);
+      });
+    };
+
+    /*
+    * Update scope values on input change
+    */
+    fieldLocationCtrl.updateModel = function() {
+      $scope.field.values[0].lat = fieldLocationCtrl.field.value.lat;
```

Comment:
If the field names of the object are the same, no need to set them individually.

You can just have `$scope.field.values[0] = fieldLocationCtrl.field.value`